azkaban-common/src/main/java/azkaban/jobExecutor/utils/process/AzkabanProcess.java

```diff
@@ -101,13 +114,15 @@ public void run() throws IOException {
       }
 
       completeLatch.countDown();
-      if (exitCode != 0) {
-        throw new ProcessFailureException(exitCode, errorGobbler.getRecentLog());
-      }
 
       // try to wait for everything to get logged out before exiting
       outputGobbler.awaitCompletion(5000);
       errorGobbler.awaitCompletion(5000);
+
+      if (exitCode != 0) {
+        throw new ProcessFailureException(exitCode, errorGobbler.getRecentLog());
```

Comment:
Should we append stdout as well ? 
While fixing Java 1.8 test cases, I observed that some failures where logged in stdout and not stderr.